api/handler.go

```diff
@@ -142,6 +152,57 @@ func getContainerInfoRequest(body io.ReadCloser) (*info.ContainerInfoRequest, er
 	return &query, nil
 }
 
+func getEventRequest(request []string) (*events.Request, bool, error) {
+	if len(request) <= 0 || len(request[0]) == 0 {
+		return events.NewAllEventsRequest(), true, nil
+	} else if len(request) > 1 {
+		return nil, false, errors.New("Should have received one url rawQuery but got multiple")
+	}
+	rawQuery := strings.Trim(request[0], "/ ")
+
+	query := events.NewRequest()
+	getEventsFromAllTime := false
+
+	urlMap, err := url.ParseQuery(rawQuery)
+	if err != nil {
+		glog.Errorf("url parse error of %v is %v", rawQuery, err)
+		return nil, false, err
+	}
+
+	valueMap := make(map[string]urlData)
```

Comment:
Ah I see, why don't we do the parsing per field? Since once we get the key we know what type we expect.